app/src/main/java/com/zulip/android/activities/LoginActivity.java

```diff
@@ -184,20 +188,31 @@ public void onSaveInstanceState(Bundle savedInstanceState){
     }
 
     private void showBackends(String httpScheme, String serverURL) {
+        // if server url does not end with "/", then append it
+        if (!serverURL.endsWith("/")) {
+            serverURL = serverURL + "/";
+        }
+
         Uri serverUri = Uri.parse(serverURL);
 
         serverUri = serverUri.buildUpon().scheme(httpScheme).build();
 
-        // if does not begin with "api.zulip.com" and if the path is empty, use "/api" as first segment in the path
-        List<String> paths = serverUri.getPathSegments();
-        if (!serverUri.getHost().startsWith("api.") && paths.isEmpty()) {
-            serverUri = serverUri.buildUpon().appendEncodedPath("api/").build();
-        }
+        // display server url with http scheme used
         serverIn.setText(serverUri.toString());
         mServerEditText.setText(serverUri.toString());
         mServerEditText.setEnabled(false);
+
+        // if server url does not end with "api/" or if the path is empty, use "/api" as last segment in the path
+        List<String> paths = serverUri.getPathSegments();
+        if (paths.isEmpty() || !paths.get(paths.size() - 1).equals("api")) {
+            serverUri = serverUri.buildUpon().appendEncodedPath("api/").build();
+        }
+
         ((ZulipApp) getApplication()).setServerURL(serverUri.toString());
 
+        // create new zulipServices object every time by setting it to null
+        getApp().setZulipServices(null);
```

Comment:
Even after successful login the ZulipServices will be null, now when AsyncGetEvents calls getZulipServices  here https://github.com/zulip/zulip-android/blob/c9855cf2b4f531b325e65fe324a362be75396553/app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java#L107

Those objects will be re initialized!